velox/type/Type.cpp

```diff
@@ -1008,6 +1008,8 @@ std::string DateType::toString(int32_t days) const {
       days);
   TimestampToStringOptions options;
   options.mode = TimestampToStringOptions::Mode::kDateOnly;
+  // Enable zero-padding for year, to ensure compliance with 'YYYY' format.
```

Comment:
There is VELOX_CHECK on L1005. When would it fire? It seems like it should be a user error, no?

Reply:
Hmm, `VELOX_USER_CHECK` does seem to make more sense there, as it would really only fail if there is issue with the Date input passed in.

Reply:
This check fires when `Timestamp::epochToUtc()` returns false, if it's not able to generate an `std::tm` type based on the input `Date` - and this only happens when the year value exceeds the bounds of the int32 field it will be stored in the `tm`, to prevent overflow.

The check in the Date constructor takes care of possible user error by validating the int32 used to represent days-since-epoch - but the possible year overflow wouldn't be exposed until we actually derive the year, as in `epochToUtc()`.